### PR TITLE
Uint64 to string in pure nim. array[char] to string fixed in vm.

### PR DESCRIPTION
**Pull request by yglukhov**
As a result, VM can now convert uint64 to string.

### CODE REVIEW

File: lib/system/repr.nim [hunk 1]
```diff
@@ -22,7 +22,22 @@ proc reprPointer(x: pointer): string {.compilerproc.} =
...
+  var i = 0
+
+  var n = x
+  while n != 0:
+    let nn = n div 10.uint64
+    buf[i] = char(n - 10.uint64 * nn + ord('0'))
+    inc i
+    n = nn
+
+  let half = int(i div 2)
```
Comment (Araq) [inline lib/system/repr.nim]: conversion to 'int' shouldn't be required.

File: lib/system/repr.nim [hunk 2]
```diff
@@ -22,7 +22,22 @@ proc reprPointer(x: pointer): string {.compilerproc.} =
...
+  # Reverse
+  for t in 0 .. < half: swap(buf[t], buf[i-t-1])
+
+  if x == 0:
```
Comment (Araq) [inline lib/system/repr.nim]: If you check for 'x == 0' you might as well move the check to the start and have the rest in an 'else'

File: lib/system/repr.nim [hunk 3]
```diff
@@ -22,7 +22,22 @@ proc reprPointer(x: pointer): string {.compilerproc.} =
 
 proc `$`(x: uint64): string =
   var buf: array [0..59, char]
-  discard c_sprintf(buf, "%llu", x)
+  var i = 0
+
+  var n = x
+  while n != 0:
+    let nn = n div 10.uint64
```
Comment (Araq) [inline lib/system/repr.nim]: convoluted, but gets rid of a div/mod pair I guess.

**Comment by yglukhov** [comment]
All fixed. You're right about temp div result. It is needed to avoid mod operation.

**Pull request merged**